src/app/pages/datasets/components/dataset-management/dataset-management.component.ts

```diff
@@ -118,34 +122,89 @@ export class DatasetsManagementComponent implements OnInit, AfterViewInit {
       });
   }
 
-  onSearch(query: string): void {
-    this.dataSource.filter(query);
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
   }
 
-  private setupTree(): void {
-    this.datasetStore.datasets$
-      .pipe(untilDestroyed(this))
+  listenForSystemdatasetConfig(): void {
```

Comment:
this is not really listen. More like `loadSystemDataset`

Reply:
should agree